sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/implementation/authentication/AccessTokensImpl.java

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.containers.containerregistry.implementation.authentication;
+
+import com.azure.containers.containerregistry.implementation.models.AcrAccessToken;
+import com.azure.containers.containerregistry.implementation.models.AcrErrorsException;
+import com.azure.core.annotation.ExpectedResponses;
+import com.azure.core.annotation.Host;
+import com.azure.core.annotation.Post;
+import com.azure.core.annotation.ServiceInterface;
+import com.azure.core.annotation.FormParam;
+import com.azure.core.annotation.HeaderParam;
+import com.azure.core.annotation.HostParam;
+import com.azure.core.annotation.UnexpectedResponseExceptionType;
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.RestProxy;
+import com.azure.core.util.Context;
+import com.azure.core.util.FluxUtil;
+import com.azure.core.util.serializer.SerializerAdapter;
+import reactor.core.publisher.Mono;
+
+/** An instance of this class provides access to all the operations defined in AccessTokensService. */
+public final class AccessTokensImpl {
```

Comment:
>public [](start = 0, length = 6)

Why do we generate these classes as public?

Reply:
I had to hand author the generated files due to the discrepancy between www-encoded form not correctly supported. Also, I could not use the builder pattern since this class is a hierarchial pattern and has only one builder. And it needed the policy that had to be added to the builder to consume this class. Unfortunately ACR's STS goes through the same rest endpoints. Let me know if there is a better way to do it.

---
In reply to: [596415910](https://github.com/Azure/azure-sdk-for-java/pull/19926#discussion_r596415910) [](ancestors = 596415910)

Reply:
These classes are generated as `public` since Java classes cannot be accessible across package boundaries without either being `protected` or `public`.

Reply:
I missed that they were in a different package.

---
In reply to: [597119525](https://github.com/Azure/azure-sdk-for-java/pull/19926#discussion_r597119525) [](ancestors = 597119525)

Reply:
@pallavit could you please file a bug in [autorest.java](https://github.com/Azure/autorest.java) repo with steps to reproduce and the expected code that should be generated? Ideally, we should not modify any generated code manually as it would be overwritten when we regenerate this code.

Reply:
There already is a bug and @jianghaolu  is aware of it.

---
In reply to: [597954987](https://github.com/Azure/azure-sdk-for-java/pull/19926#discussion_r597954987) [](ancestors = 597954987)